### DISCUSSION TITLE
We should rethink our current logs

### DISCUSSION THREAD

**Discussion by J0sueTM** [moclojer/moclojer-app#363] [Ideas]
While testing during development of the task #361, I noted that a single user usage of moclojer raises around 300 logs every 5 minutes or so. Most of these logs were debug info or had duplicated or unnecessary information. These include:

- request data, which contained way more user identifying info than necessary
- debug logs that were probably necessary in dev mode, but not in prod env

@Felipe-gsilva @avelino @matheusfrancisco I ask you guys to give a look at the logs in the opensearch dashboard. It's under `moclojer-api-test-logs` index.

What do you guys think? This should probably be taken seriously before launch. If it's already overwhelming for a single local user, imagine with hundreds, or even thousands of simultaneous users.

**Comment by Felipe-gsilva** [comment]
there are at least 4/5 fields we could just remove from every log

let me also ask, when will this parent uuid be filled? because it is set to nil in every log we got until now

**Reply by J0sueTM** [reply]
Are you sure? The parent field is about the parent trace. Most currently don't have data because I'm still updating our codebase. But some of them do have the field related to the parent trace, you probably wasn't able to find one while using OpenSearch :smile:

**Reply by J0sueTM** [reply]
Also, the density of the logs aren't my main worry. I'm specifically talking about the what information we're sending, and how many logs we're aggregating, not how big the logs are. They sure impact a lot, like unnecessary or duplicated stuff in the data field (which btw we can update easily in our own implementation), but as I said before, aren't the main worry.

**Reply by Felipe-gsilva** [reply]
I was just unfortunate, I think. The 9 hits from source = moclojer-api-test-logs seems to have empty parent/uid :smiling_face_with_tear:

**Reply by Felipe-gsilva** [reply]
> Also, the density of the logs aren't my main worry. I'm specifically talking about the what information we're sending, and how many logs we're aggregating, not how big the logs are. They sure impact a lot, like unnecessary or duplicated stuff in the data field (which btw we can update easily in our own implementation), but as I said before, aren't the main worry.

fair enough :+1:

**Reply by J0sueTM** [reply]
> 9 hits from source

9 hits? There are 300+ hits in the index I mentioned

**Reply by Felipe-gsilva** [reply]
yeah, thats strange, I only get 9 hits from this source

**Reply by Felipe-gsilva** [reply]
![image](https://github.com/user-attachments/assets/4d318dd0-f955-4dd4-b1c8-28c7e2382eba)
take a look and check if Im not crazy LOL

**Reply by J0sueTM** [reply]
Wtf????? For some reason it went down to 9 hits lmao. Maybe it parsed them and returned only the unique ones? I'm still figuring it out

**Reply by J0sueTM** [reply]
![image](https://github.com/user-attachments/assets/948d7fa1-9d7f-4186-9677-9c625cae9305)

@Felipe-gsilva The index manager says there are really 300 documents, my suggestion is that opensearch somehow really condensed duplicated documents, or showed only those where all data isn't null... I'm just shooting in the dark at this point.

**Reply by Felipe-gsilva** [reply]
well, if it auto fix our problem, we should see a way to configure this compression or whatever this is

**Reply by J0sueTM** [reply]
@Felipe-gsilva pelo que percebi, ele so guarda os traces, entao ele deve estar entendendo alguma coisa. O restante das mensagens estao sim sendo guardadas, mas quando fazemos query, so estamos pegando os logs de trace.

**Reply by J0sueTM** [reply]
@Felipe-gsilva Entendi o que esta acontecendo:

O OpenSearch tenta automaticamente agrupar logs que tem um elemento em comum. Por algum motivo ele chutou que o elemento em comum seria o parent/uid, e o tempo em comum seria o run-nsecs, o que deixava entao so os traces aparecerem, pois so eles tem os 2 valores nao nulos.

Alterei pra msg_ como o elemento em comum, e o inst como tempo em padrao. Agora voltou os 300 logs.

**Comment by J0sueTM** [comment]
![image](https://github.com/user-attachments/assets/2a205268-6b85-4829-a8a0-6dce955d2bf2)

Num sample de 300 logs, praticamente 1 terco sao logs de execucao de queries SQL. Como estamos utilizando de traces pra identificar as funcoes de DB, podemos desativar os logs por meio do next.jdbc

**Reply by Felipe-gsilva** [reply]
looks fine to have up to 180 as a normal flow of data, since the opensearch will already compress them :fire: :rocket:
